debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -245,6 +224,7 @@ else if (oplogStart.getTime() > 0) {
                 if (event != null) {
                     LOGGER.trace("Arrived Change Stream event: {}", event);
 
+                    // TODO: Should this be removed as part of DBZ-5102?
```

Comment:
This now appears superfluous. However, there could be some edge cases for which the collection name doesn't show up in the change event. For example, the [dropDatabase]( https://www.mongodb.com/docs/manual/reference/change-events/dropDatabase/#mongodb-data-dropDatabase) event doesn't contain a `ns.coll` field, which would not be excluded by the new server-side filter.

Reply:
Yes, I would say so. The pipeline setup now guarantees that only events on desired databases are reported by database.